odoo/addons/base/models/ir_qweb.py

```diff
@@ -613,16 +665,145 @@ def _render(self, template, values=None, **options):
             raise ValueError(f'values[{T_CALL_SLOT}] should be unset when call the _render method and only set into the template.')
 
         irQweb = self.with_context(**options)._prepare_environment(values)
+        irQweb = irQweb.with_context(
+            # List of generated and/or used functions, used for optimal performance
+            __qweb_loaded_functions={},
+            # List of codes generated during compilation. It is mainly used for debugging and displaying error messages.
+            __qweb_loaded_codes={},
+            # Reference to the last node being compiled. It is mainly used for debugging and displaying error messages.
+            _qweb_error_path_xml=[None, None],
+        )
 
         safe_eval.check_values(values)
 
-        template_functions, def_name = irQweb._compile(template)
-        render_template = template_functions[def_name]
-        rendering = render_template(irQweb, values)
-        result = ''.join(rendering)
+        root_values = values.copy()
+        values['__qweb_root_values'] = root_values['__qweb_root_values'] = root_values
+
+        result = ''.join(irQweb._render_iterall(template, None, values))
 
         return Markup(result)
 
+    def _render_iterall(self, view_ref, method, values):
+        """ Iterate over the generator method.
+            Generator elements are a str or a tuple (ref, function_name, default_values)
+        """
+        root_values = values['__qweb_root_values']
+        loaded_functions = self.env.context['__qweb_loaded_functions']
+        irQweb = self
+        stack = [{
```

Comment:
A [namedtuple](https://docs.python.org/3/library/collections.html#collections.namedtuple) would be more efficient than a dict since you create a lot of these.

For readability, `item` returned by the iterator could be a named tuple too. But maybe it is too much changes.

Reply:
In the new version is
```py
QwebContent = namedtuple('QwebContent', 'context view_ref method values scope directive log')
```
But I would like to find a best name. The name must reflect that the content can be loaded from this information. It's not realy the content.